frb_codegen/src/main.rs

```diff
@@ -26,3 +26,195 @@ fn main() -> anyhow::Result<()> {
     info!("Now go and use it :)");
     Ok(())
 }
+
+mod tests {
+    use std::fs;
+
+    use lazy_static::lazy_static;
+    use lib_flutter_rust_bridge_codegen::init_logger;
+
+    lazy_static! {
+        static ref LOGGER: () = init_logger(".", true).unwrap();
+    }
+
+    // VS Code runs in frb_codegen with "Run test" and flutter_rust_bridge with "Debug test" >_>
+    #[allow(dead_code)]
+    fn set_dir() {
+        if let Ok(metadata) = fs::metadata("frb_codegen") {
+            if metadata.is_dir() {
+                std::env::set_current_dir("frb_codegen").unwrap();
+            }
+        }
+    }
+
+    #[test]
+    fn pure_dart() {
+        // `cargo build` can fail because frb_example/pure_dart makes testing a build step! *gasp*
+        // If you want to get `cargo build` working again and investigate why pure_dart fails:
+        // 1. mv ../frb_example/pure_dart/rust/build.rs ../rb_example/pure_dart/rust/_build.rs
+        // 2. Run this test in the debugger - which is a copy of build.rs (but more complete)
```

Comment:
So, I wonder whether the `mv` step necessary? In this `fn pure_dart`, we are firstly calling `frb_codegen` *function*, and then call `cargo build` etc. So, if the build.rs fails (inside cargo build), then our frb_codegen *function* should already fail, so cargo build is never called. Thus, `mv` step seems unneeded.

Reply:
It depends on the workflow. I'm using VS Code. The official rust-analyzer plugin will run `cargo-build` before it even starts the test. If `build.rs` fails, it won't even attempt to run the test. That's the catch-22.